src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test;Password=test;"",
+    ""options"": {
+      ""set-session-context"": true
+    }
+  },
+  ""runtime"": {
+    ""rest"": {
+      ""enabled"": " + initialRestEnabled.ToString().ToLower() + @",
+      ""path"": """ + initialRestPath + @"""
+    },
+    ""graphql"": {
+      ""enabled"": " + initialGQLEnabled.ToString().ToLower() + @",
+      ""path"": """ + initialGQLPath + @""",
+      ""allow-introspection"": " + initialGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + initialMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string updatedConfig = @"
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test;Password=test;"",
+    ""options"": {
+      ""set-session-context"": true
+    }
+  },
+  ""runtime"": {
+    ""rest"": {
+      ""enabled"": " + updatedRestEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedRestPath + @"""
+    },
+    ""graphql"": {
+      ""enabled"": " + updatedGQLEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedGQLPath + @""",
+      ""allow-introspection"": " + updatedGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + updatedMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string configName = "config.json";
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            FileSystem fileSystem = new();
+            fileSystem.File.WriteAllText(configName, initialConfig);
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configName, string.Empty);
+            RuntimeConfigProvider configProvider = new(configLoader);
+            // Must GetConfig() to start file watching
+            RuntimeConfig runtimeConfig = configProvider.GetConfig();
+            Assert.IsNotNull(runtimeConfig);
+            // Simulate change to the config file
+            fileSystem.File.WriteAllText(configName, updatedConfig);
+            // Give file watcher enough time to see the change
+            System.Threading.Thread.Sleep(1000);
```

Comment:
is it the recomended sleep time for file watcher? what if this is not enough?

Reply:
there isn't really a recommended sleep time for this case, but 1000ms is well more than enough time to pick up the change and it doesnt significantly affect the time to complete the test.